### PR TITLE
Use chrome APIs to trigger first-run page

### LINKED ISSUE #88: Use the chrome.runtime api onInstalled event to launch the firstrun page
There is an [event](https://developer.chrome.com/extensions/runtime#event-onInstalled) provided by the chrome.runtime api to determine when an extension is updated or installed.  Currently we use stored values in LocalStorage to determine this.  We should update [firstrun.js](https://github.com/privly/privly-chrome/blob/master/javascripts/background_scripts/first_run.js) to use this API instead of maintaining this functionality ourselves.

### PR DESCRIPTION
**Pull request by suryatech**
Fixes #88
Opens first-run page on new installations.
Minor changes in test files.

### CODE REVIEW

File: javascripts/background_scripts/first_run.js [hunk 1]
```diff
@@ -52,14 +52,25 @@ var firstRun = {
...
+
+	    chrome.windows.getLastFocused(function(w){
+		      if(w.id){
+			      chrome.tabs.create({
+				      url: page,
+				      active: true,
+				      windowId: w.id
+			      });
+		      }else{
+			      console.log("Cannot get active window");
```
Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: should be console.warn or error, no console.logs in master.

Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: You also have a ton of spacing here. Make sure you don't indent more than two spaces per level of nesting.

File: javascripts/background_scripts/test_loader.js [hunk 2]
```diff
@@ -26,7 +26,7 @@ function loadJs(filename){
 function getMetaValue(metaName){
   var metas = document.getElementsByTagName("meta");
   for (var i = 0; i < metas.length; i++){
-    if(metas[i].getAttribute('name') == metaName){
+    if(metas[i].getAttribute('name') === metaName){
```
Comment (smcgregor) [inline javascripts/background_scripts/test_loader.js]: The testing parts should be on a different commit or not included at all. The way we are running unit tests outside privly-applications is also outdated -- as in they are not integrated with Karma. This is something we need to open an issue for...

File: javascripts/background_scripts/first_run.js [hunk 3]
```diff
@@ -86,13 +97,11 @@ var firstRun = {
...
-    var runningVersion = firstRun.getPrivlyVersion();
-    var lastRunVersion = firstRun.getStoredVersion();
-
-    if (lastRunVersion === null || runningVersion !== lastRunVersion ) {
-      firstRun.firstRun();
-      firstRun.updateVersion(runningVersion);
-    }
+    chrome.runtime.onInstalled.addListener(function(details){
+      if(details.reason === 'install' || details.reason === 'update'){
+				firstRun.firstRun();
```
Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: This is excellent.

**Comment by smcgregor** [comment]
Mostly stylistic changes need to be made.

This pull request shows we _really_ need to update he unit test infrastructure across the non-privly-applications parts of the code base.

File: javascripts/background_scripts/first_run.js [hunk 4]
```diff
@@ -74,25 +83,23 @@ var firstRun = {
...
+      ls.setItem("glyph_color", Math.floor(Math.random() * 16777215).toString(16));
       var glyph_cells = ((Math.random() < 0.5) ? "false" : "true");
-      for(var i = 0; i < 14; i++) {
+      for (var i = 0; i < 14; i++) {
         glyph_cells += "," + ((Math.random() < 0.5) ? "false" : "true");
       }
       ls.setItem("glyph_cells", glyph_cells);
     }
 
-    var runningVersion = firstRun.getPrivlyVersion();
```
Comment (smcgregor) [inline javascripts/background_scripts/first_run.js]: Always set the version here and remove the check for the stored version in firstRun.firstRun. Much of this version comparison logic is not necessary if we use the Chrome API to open the firstRun page.

**Comment by smcgregor** [comment]
If you don't comment about the changes you made to the PR, it won't make its way back into the review queue.

**Comment by smcgregor** [comment]
You added commits a while back. Should I review this? What did you change? Summaries help.

**Comment by suryatech** [comment]
@smcgregor 

This [commit](https://github.com/privly/privly-chrome/commit/0f821f750e5c2744b1819eed1eb79f05b076364d) shows that chrome APIs are already in place to trigger the firstRun page (albeit using a setTimeout hack)

I don't see a point working on this.
